.github/workflows/daily-wpt-fyi.yml

```diff
@@ -0,0 +1,110 @@
+# This workflow runs every night and tests various versions of Node.js
+# (main branch build, current, and two latest LTS release lines) against the
+# `epochs/daily` branch of WPT.
+
+name: Daily WPT report
+
+on:
+  workflow_dispatch:
+  schedule:
+    # This is 20 minutes after `epochs/daily` branch is triggered to be created
+    # in WPT repo.
+    # https://github.com/web-platform-tests/wpt/blob/master/.github/workflows/epochs.yml
+    - cron: 30 0 * * *
+
+env:
+  PYTHON_VERSION: '3.11'
+
+permissions:
+  contents: read
+
+jobs:
+  generate:
+    strategy:
+      matrix:
+        ref:
+          - main
+          - current
+          - lts/*
+          - lts/-1
+      fail-fast: false
+    runs-on: ubuntu-latest
+    continue-on-error: true
+    steps:
+      - name: Set up Python ${{ env.PYTHON_VERSION }}
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - name: Environment Information
+        run: npx envinfo
+
+      # checkout main & build
+      - name: Checkout ${{ matrix.ref }} branch
+        if: matrix.ref == 'main'
+        uses: actions/checkout@v3
+        with:
+          ref: ${{ matrix.ref }}
+          persist-credentials: false
+      - name: Build Node.js
+        if: matrix.ref == 'main'
+        run: make build-ci -j2 V=1 CONFIG_FLAGS="--error-on-warn"
+
+      # or install a version and checkout
+      - name: Install Node.js
+        if: matrix.ref != 'main'
+        id: setup-node
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.ref }}
+      - name: Checkout ${{ steps.setup-node.outputs.node-version }}
+        uses: actions/checkout@v3
+        if: matrix.ref != 'main'
+        with:
+          persist-credentials: false
+          ref: ${{ steps.setup-node.outputs.node-version }}
+      - name: Set env.NODE
+        if: matrix.ref != 'main'
+        run: echo "NODE=$(which node)" >> $GITHUB_ENV
+
+      # replace checked out WPT with the synchronized branch
+      - name: Remove stale WPT
+        run: rm -rf wpt
+        working-directory: test/fixtures
+      - name: Checkout epochs/daily WPT
```

Comment:
Is there a way to reuse this step for each matrix element, instead of checking out for each?

Reply:
Not that I'm aware of.